charts/vector/templates/_helpers.tpl

```diff
@@ -241,9 +241,95 @@ Print the `url` option for the Vector command.
+{{ print "\\" }}
+        --url {{ printf "http://%s/graphql" $.Values.customConfig.api.address }}
       {{- end }}
     {{- end }}
   {{- end }}
 {{- end }}
 
+{{/*
+Configuring Datadog Agents to forward to Vector.
+This is really alpha level, and should be refactored
+to be more generally usable for other components.
+*/}}
+{{- define "_configure.datadog" -}}
+{{- $hasSourceDatadogAgent := false }}
+{{- $sourceDatadogAgentPort := "" }}
+{{- $hasTls := "" }}
+{{- $protocol := "http" }}
+{{- range $componentKind, $configs := .Values.customConfig }}
+  {{- if eq $componentKind "sources" }}
+    {{- range $componentId, $componentConfig := $configs }}
+      {{- if eq (get $componentConfig "type") "datadog_agent" }}
+	{{- $hasSourceDatadogAgent = true }}
+        {{- $sourceDatadogAgentPort = mustRegexFind "[0-9]+$" (get $componentConfig "address") }}
+	{{- if (hasKey $componentConfig "tls") }}
+	  {{- $tlsOpts := get $componentConfig "tls" }}
+	  {{- $hasTls = get $tlsOpts "enabled" }}
+	  {{- if $hasTls }}{{ $protocol = "https" }}{{ end }}
+	{{- end }}
+      {{- end }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+{{- if or (not .Values.customConfig) (and .Values.customConfig $hasSourceDatadogAgent) }}
+{{- template "_divider" }}
+
+{{ print "\033[36;1mdatadog_agent:\033[0m" }}
+
+To forward logs from Datadog Agents deployed with the {{ include "_fmt.yellow" "datadog" }} Helm chart,
+include the following in the {{ include "_fmt.yellow" "values.yaml" }} for your {{ include "_fmt.yellow" "datadog" }} chart:
+
+For Datadog Agents version {{ include "_fmt.yellow" "7.35" }}/{{ include "_fmt.yellow" "6.35" }} or greater:
+
+{{ include "_fmt.blue" "datadog:" }}
+  {{ include "_fmt.blue" "containerExclude:" }} "name:vector"
+  {{ include "_fmt.blue" "logs:" }}
+    {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+    {{ include "_fmt.blue" "containerCollectAll:" }} {{ include "_fmt.yellow" "true" }}
+{{ include "_fmt.blue" "agents:" }}
+  {{ include "_fmt.blue" "useConfigMap:" }} {{ include "_fmt.yellow" "true" }}
+  {{ include "_fmt.blue" "customAgentConfig:" }}
+    {{ include "_fmt.blue" "kubelet_tls_verify:" }} {{ include "_fmt.yellow" "false" }}
+    {{ include "_fmt.blue" "vector:" }}
+      {{ include "_fmt.blue" "logs:" }}
+        {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+        {{- if not .Values.haproxy.enabled }}
+        {{ include "_fmt.blue" "url:" }} "{{ $protocol }}://{{ include "vector.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
+        {{- else }}
+        {{ include "_fmt.blue" "url:" }} "{{ $protocol }}://{{ include "haproxy.fullname" $ }}.{{ $.Release.Namespace }}:{{ $sourceDatadogAgentPort | default "8282" }}"
+        {{- end }}
+      {{ include "_fmt.blue" "metrics:" }}
+        {{ include "_fmt.blue" "enabled:" }} {{ include "_fmt.yellow" "true" }}
+        {{- if not .Values.haproxy.enabled }}
```

Comment:
I think it would be clearer to not have the not and swap the if and else blocks around..

```suggestion
        {{- if .Values.haproxy.enabled }}
```